### PR TITLE
[sleepy] Rename sleepy params per spec.

### LINKED ISSUE #17220: [sleepy] Update CRI/CRA naming per recent spec updates.
#### Problem
CRI/CRA params are in process of being updated in recent spec.

#### Proposed Solution
Use the final names: SII, SAI.

### PR DESCRIPTION
**Pull request by turon**
#### Problem
Update sleepy parameters per https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/5092
Fixes https://github.com/project-chip/connectedhomeip/issues/17220

#### Change overview
Initial rename of CRA/CRI discovery params --> SAI/SII.

#### Testing
CI

### CODE REVIEW

File: src/app/tests/suites/certification/Test_TC_SC_4_5.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ tests:
       verification:
           "1. On the raspi controller, publish matter service, using below
           command $avahi-publish-service 87E1B004E235A130-8FC7772401CD0696
-          _matter._tcp 22222 CRI=3000 CRA=4000 T=0"
+          _matter._tcp 22222 SII=3000 SAI=4000 T=0"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_SC_4_5.yaml:29]: Having a SAI > SII seems... odd.  It sort of made sense maybe when these were retry intervals, but now that these are supposed to be the sleep periods, this makes no sense.

File: src/lib/dnssd/Resolver.h [hunk 2]
```diff
@@ -68,7 +68,7 @@ struct ResolvedNodeData
 
     bool IsDeviceTreatedAsSleepy(const ReliableMessageProtocolConfig * defaultMRPConfig) const
     {
-        // If either retry interval (Idle - CRI, Active - CRA) has a value and that value is greater
+        // If either retry interval (Idle - SII, Active - SAI) has a value and that value is greater
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/Resolver.h:71]: They're not retry intervals anymore.  They're sleep intervals...

**Review by tcarmelveilleux — DISMISSED** [review]
No changes requested, but putting this merge gate until the spec PR is merged. We should not implement in SDK until spec PR merges.

**Review by msandstedt — APPROVED** [review]

File: src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp [hunk 3]
```diff
@@ -228,7 +228,7 @@ class AdvertiserMinMdns : public ServiceAdvertiser,
                     mrp.mIdleRetransTimeout = kMaxRetryInterval;
                 }
                 size_t writtenCharactersNumber = snprintf(storage.mrpRetryIntervalIdleBuf, sizeof(storage.mrpRetryIntervalIdleBuf),
-                                                          "CRI=%" PRIu32, mrp.mIdleRetransTimeout.count());
+                                                          "SII=%" PRIu32, mrp.mIdleRetransTimeout.count());
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp:231]: I assume there will be a followup to rename mIdleRetransTimeout to something that more closely matches what this actually is?

File: src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp [hunk 4]
```diff
@@ -244,7 +244,7 @@ class AdvertiserMinMdns : public ServiceAdvertiser,
                     mrp.mActiveRetransTimeout = kMaxRetryInterval;
                 }
                 size_t writtenCharactersNumber =
-                    snprintf(storage.mrpRetryIntervalActiveBuf, sizeof(storage.mrpRetryIntervalActiveBuf), "CRA=%" PRIu32,
+                    snprintf(storage.mrpRetryIntervalActiveBuf, sizeof(storage.mrpRetryIntervalActiveBuf), "SAI=%" PRIu32,
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp:247]: Similar for mActiveRetransTimeout.

File: src/lib/dnssd/minimal_mdns/tests/TestAdvertiser.cpp [hunk 5]
```diff
@@ -89,7 +89,7 @@ OperationalAdvertisingParameters operationalParams5 =
     OperationalAdvertisingParameters().SetPeerId(kPeerId5).SetMac(ByteSpan(kMac)).SetPort(CHIP_PORT).EnableIpV4(true);
 OperationalAdvertisingParameters operationalParams6 =
     OperationalAdvertisingParameters().SetPeerId(kPeerId6).SetMac(ByteSpan(kMac)).SetPort(CHIP_PORT).EnableIpV4(true);
-const QNamePart txtOperational1Parts[]  = { "CRI=32", "CRA=33", "T=0" };
+const QNamePart txtOperational1Parts[]  = { "SII=32", "SAI=33", "T=0" };
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/minimal_mdns/tests/TestAdvertiser.cpp:92]: Followup: again, SII < SAI does not make sense...

File: src/lib/dnssd/platform/tests/TestPlatform.cpp [hunk 6]
```diff
@@ -59,8 +59,8 @@ test::ExpectedCall operationalCall2 = test::ExpectedCall()
                                           .SetInstanceName("5555666677778888-1212343456567878")
                                           .SetHostName(host)
                                           .AddSubtype("_I5555666677778888")
-                                          .AddTxt("CRI", "32")
-                                          .AddTxt("CRA", "33")
+                                          .AddTxt("SII", "32")
+                                          .AddTxt("SAI", "33")
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/platform/tests/TestPlatform.cpp:63]: Again, SII < SAI does not make sense.  Followup.

File: src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp (hunk 3 above)
Comment (turon) [reply src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp:231]: Yes, this PR is focused on cleanly renaming the Discovery names only to minimize impact on CSG.

Comment (turon) [reply src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp:231]: #17999

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by turon** [comment]
> No changes requested, but putting this merge gate until the spec PR is merged. We should not implement in SDK until spec PR merges.

Spec changes merged: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/5092

**Review by andy31415 — APPROVED** [review]

**Pull request merged**